src/query/procedure/mg_procedure_impl.cpp

```diff
@@ -319,6 +320,63 @@ mgp_value_type FromTypedValueType(memgraph::query::TypedValue::Type type) {
 }
 }  // namespace
 
+bool MgpGraphIsTransactional(const mgp_graph &graph) noexcept {
+  auto storageMode = memgraph::storage::StorageMode::IN_MEMORY_TRANSACTIONAL;
+  std::visit(memgraph::utils::Overloaded{
+                 [&storageMode](memgraph::query::DbAccessor *dbAcc) { storageMode = dbAcc->GetStorageMode(); },
+                 [&storageMode](memgraph::query::SubgraphDbAccessor *dbAcc) { storageMode = dbAcc->GetStorageMode(); }},
+             graph.impl);
+  return storageMode != memgraph::storage::StorageMode::IN_MEMORY_ANALYTICAL;
+}
+
+bool IsDeleted(mgp_vertex *vertex) { return vertex->getImpl().impl_.vertex_->deleted; }
```

Comment:
Did you think about case when you call procedure on trigger, and node was deleted but for_deleted_ which is used only for triggers is marked as true, which means that node can actually be used here, and you can work with it, so this will be some kind of edge case.

Reply:
So to expand imagine this:

STORAGE MODE IN_MEMORY_ANALYTICAL;

CREATE TRIGGER my_trigger on () DELETE CALL my_proc.count_deleted()

MATCH (n) DETACH DELETE n;

I think this will create scenario where node will be marked as deleted, but for_deleted_ flag inside VertexAccessor will be true, meaning we can work with it now inside that procedure

Reply:
The sole purpose of the deletion checks is to avoid returning deleted objects as results. Triggers operating on deleted objects can still use them, as long as they don’t return them (this is identical to the present behavior).

Reply:
I’ll add a test case for this, thanks for the heads-up!